### PR TITLE
rootless: check that / is mounted as shared

### PR DESCRIPTION
**Pull request by giuseppe**
if the root mount '/' is not mounted as MS_SHARED, print a
warning, otherwise new mounts that are created in the host won't be
propagated to the rootless mount namespace.

Closes: https://github.com/containers/podman/issues/10946

[NO TESTS NEEDED]

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM
@containers/podman-maintainers PTAL

File: pkg/rootless/rootless_linux.go [hunk 1]
```diff
@@ -235,6 +237,24 @@ func becomeRootInUserNS(pausePid, fileToRead string, fileOutput *os.File) (_ boo
...
+			isShared := false
+			for _, o := range strings.Split(m.Optional, ",") {
+				if strings.HasPrefix(o, "shared:") {
+					isShared = true
+					break
+				}
+			}
+			if !isShared {
+				logrus.Warningf("%q is not a shared mount, this could cause issues or missing mounts with rootless containers", m.Mountpoint)
+			}
```
Comment (Luap99) [inline pkg/rootless/rootless_linux.go:254]: Should there be a break here? There is no point in looping over the remaining mount points.

Comment (giuseppe) [reply pkg/rootless/rootless_linux.go:254]: thanks for spotting it.  Fixed and pushed a new version

**Comment by rhatdan** [comment]
Is this going to warn all the time?  If I have a system that is built with no / as private, will podman constantly complain?

**Comment by giuseppe** [comment]
> Is this going to warn all the time? If I have a system that is built with no / as private, will podman constantly complain?

just the first time when we create the user+mount namespace.

**Comment by rhatdan** [comment]
so the first time after use login, everytime.

**Comment by rhatdan** [comment]
/lgtm
Might need an option eventually to turn this off,  we shall see.

**Pull request merged**